ui-components/src/helpers/filters.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  ListingFilterParams,
+  ListingFilterKeys,
+  EnumListingFilterParamsComparison,
+} from "@bloom-housing/backend-core/types"
+import { ParsedUrlQuery } from "querystring"
+
+function getComparisonForFilter(filterKey: ListingFilterKeys) {
+  switch (filterKey) {
+    case ListingFilterKeys.name:
+      return EnumListingFilterParamsComparison["="]
+    case ListingFilterKeys.neighborhood:
+      return EnumListingFilterParamsComparison["="]
+    case ListingFilterKeys.status:
+      return EnumListingFilterParamsComparison["="]
+  }
+}
+
+export function encodeToBackendFilterString(filterParams: ListingFilterParams) {
+  let queryString = ""
+  for (const filterType in filterParams) {
+    if (filterType in ListingFilterKeys) {
+      const comparison = getComparisonForFilter(ListingFilterKeys[filterType])
+      queryString += `&filter[$comparison]=${comparison}&filter[${filterType}]=${filterParams[filterType]}`
+    }
+  }
+  return queryString
+}
+
+export function encodeToFrontendFilterString(filterParams: ListingFilterParams) {
+  let queryString = ""
+  for (const filterType in filterParams) {
```

Comment:
just for fun:
If you wanted to do this in less iterative way, I _think_ could be something like
```
const queryString = Object.entries(filterParams)
  .filter((filterType, value) => filterType in ListingsFilterKeys && value && value != "")
  .map((filterType, value) => `&${filterType}=${value}`)
  .join()
```

Reply:
Oh! could even move the `&` into the `.join()` call like `.join('&')` so there's not one on the front (although maybe that's necessary if there's limits and pages and stuff)

Reply:
I'm going to look into how Object.entries is transformed in Bloom (since it's not supported in IE), but will leave this for now